### ISSUE TITLE
Add support for CRediT standard for contributor attribution

### ISSUE THREAD

**Issue by ctgraham** [pkp/pkp-lib#857]
Update July 17th 2025

Requirements:
The functionality from the existing plugin https://github.com/pkp/credit has been moved to the core:
There is a possibility to define CRediT roles (s. https://credit.niso.org/) and their degrees (‘lead’, ‘equal’, or ‘supporting’) for authors/contributors. So for each contributor one or several CRediT roles can be assigned and optionally degree for those roles -- all optional. The role and degree assignment happens in the contributor form, when adding or editing a contributor. 
The author's CRediT roles and degrees, if they exist, should be displayed on the article landing page. They should also be considered in exports/indexing. Currently we can consider them only in our JATS export, the other services, for example Crossref, still do not support it (but they are working on it, so soon...).

Implementation notes: 
We use https://github.com/contributorshipcollaboration/credit-translation (as we have in the plugin) to get the list of roles and their translations. The role's URLs are read from https://github.com/contributorshipcollaboration/credit-translation/blob/main/translations/en.json and are saved in the DB table credit_roles on installation or upgrade to 3.6. The names and their translations are retrieved from the .json files, and are cached for journal/press/server supported locales.

Screenshots:

CRediT area at add/edit contributor:
<img width="4749" height="465" alt="Image" src="https://github.com/user-attachments/assets/4fedd898-cb2d-4128-b689-7c56f12d5254" />

Display CRediT roles on article landing page:
<img width="250" alt="Image" src="https://github.com/user-attachments/assets/7a7480d9-d496-47e0-8bee-e59a706f06bc" />

<hr>

**Describe the problem you would like to solve**
There are many kinds of contributions to a scholarly work. OJS, OMP and OPS have no way to capture these distinct roles in a way that is consistent across the industry.

**Describe the solution you'd like**
Adopt support for the [CRediT standard](https://www.niso.org/standards-committees/credit) for contributor attribution. This standard has been approved by NISO and identifies 14 different kinds of contributions.

More details about the roles can be found in [this comment](https://github.com/pkp/pkp-lib/issues/857#issuecomment-550566116) and a further note on the workflow can be found in [this comment](https://github.com/pkp/pkp-lib/issues/857#issuecomment-759434707).

**Who is asking for this feature?**
CRediT has been approved by NISO but is not yet widely used.

**Additional information**
Below is the initial feature request posted by @ctgraham:

> Are we interfacing with CASRAI's work on enabling recognition of individual's roles within research output creation?
>
> http://casrai.org/CRediT
> 
> This seems like an important future-oriented standard.

<hr>

PRs to implement CRediT support in core:
- https://github.com/pkp/pkp-lib/pull/11639
- https://github.com/pkp/ui-library/pull/661
- https://github.com/pkp/jatsTemplate/pull/90
- https://github.com/pkp/ojs/pull/4991
- https://github.com/pkp/omp/pull/2059
- https://github.com/pkp/ops/pull/1046

UI fixes:
- https://github.com/pkp/ui-library/pull/677
- https://github.com/pkp/ui-library/commit/fc572c85815db06718bdbdf390b7e60aab6f5b12

~~PKP: https://github.com/pkp/pkp-lib/pull/11507~~
~~UI: https://github.com/pkp/ui-library/pull/644~~
~~OJS: https://github.com/pkp/ojs/pull/4930~~
~~OMP: https://github.com/pkp/omp/pull/2055~~
~~OPS: https://github.com/pkp/ops/pull/1042~~
~~jatsTemplate: https://github.com/pkp/jatsTemplate/pull/89~~

**Comment by asmecher** [comment]
Thanks, @ctgraham. Were you able to find any concrete documentation for the standard?

**Comment by ctgraham** [comment]
I don't know there is much "concrete" at this point aside from the [proposed taxonomy](http://dictionary.casrai.org/Contributor_Roles).  There appears to be quite a bit of (too much?) implementation flexibility: 

> **Do you anticipate this being a mandatory taxonomy?** No, the taxonomy is envisioned as voluntary and each system would choose whether and how to implement. We do see value in the entire community eventually agreeing on a single set of terms, but at a pace appropriate for each stakeholder.
> **Would this taxonomy be implemented as a drop-down box in software?** That is one potential mode of implementation during data entry. We envision multiple ways of implementing the taxonomy – suited to each implementation. But we hope the content of the implementations can be common and standardized.

http://dictionary.casrai.org/Contributor_Roles#Frequently_Asked_Questions

The Interest Group might be a good connection for PKP (or Development Partners).

Mirroring the functionality of the Aries Systems Corp implementation seems like a valuable and tangible goal within the current product:
> * The ability to attribute one or more ‘Contributor Roles’ to each Author of a submission
> * The ability to identify the degree to which a particular contributor was involved, i.e. one of “Lead”, “Supporting”, “Equal”
> * The ability to configure the collection of Contributor Roles per Article Type, as either optional or required.

http://casrai.org/CRediT#Aries_Systems

**Comment by fgnievinski** [comment]
I think a good start would be for the submission form to offer checkboxes with the 14 contributor roles, with hover-on definitions (listed below).

See how this was implemented in other manuscript submission systems:
https://wkauthorservices.editage.com/resources/author-resource-review/2018/may-2018.html
https://journals.plos.org/plosone/s/authorship

**Contributor Roles Defined:**
- Conceptualization – Ideas; formulation or evolution of overarching research goals and aims.
- Data curation – Management activities to annotate (produce metadata), scrub data and maintain research data (including software code, where it is necessary for interpreting the data itself) for initial use and later re-use.
- Formal analysis – Application of statistical, mathematical, computational, or other formal techniques to analyze or synthesize study data.
- Funding acquisition ​- Acquisition of the financial support for the project leading to this publication.
- Investigation – ​Conducting a research and investigation process, specifically performing the experiments, or data/evidence collection.
- Methodology – Development or design of methodology; creation of models.
- Project administration – Management and coordination responsibility for the research activity planning and execution.
- Resources – Provision of study materials, reagents, materials, patients, laboratory samples, animals, instrumentation, computing resources, or other analysis tools.
- Software – Programming, software development; designing computer programs; implementation of the computer code and supporting algorithms; testing of existing code components.
- Supervision – Oversight and leadership responsibility for the research activity planning and execution, including mentorship external to the core team.
- Validation – Verification, whether as a part of the activity or separate, of the overall replication/reproducibility of results/experiments and other research outputs.
- Visualization – Preparation, creation and/or presentation of the published work, specifically visualization/data presentation.
- Writing – original draft – ​Preparation, creation and/or presentation of the published work, specifically writing the initial draft (including substantive translation).
- Writing – review & editing – Preparation, creation and/or presentation of the published work by those from the original research group, specifically critical review, commentary or revision – including pre- or post-publication stages.

**Comment by ctgraham** [comment]
Interesting question from [emolls](https://forum.pkp.sfu.ca/t/credit-integration-with-ojs/43824/3?u=ctgraham) and [Miguel_Oliveira_Jr](https://forum.pkp.sfu.ca/t/multiple-roles-for-coauthors/57740?u=ctgraham) in the forum: Can contributor-role mapping be directly tied to OJS roles for Authors?  Are there implications in de-coupling user roles from author (sub)roles?  Are there implications in allowing an author to have many roles?

**Comment by asmecher** [comment]
@ctgraham, the submission step 1 includes a selection of what user group they are operating in when making a new submission. This list includes all "author" (ROLE_ID_AUTHOR) and "manager" (ROLE_ID_MANAGER) roles that the user has in the journal. If there is only one available selection (this is most commonly the case for journal authors) it is automatically chosen and the drop-down list is hidden. If they have no available roles, and author self-registration is enabled, then the default "author" role is automatically assigned and used behind the scenes.

Long story short: a lot of the necessary infrastructure in supporting different forms of contribution is already in place using User Groups. Mappings e.g. to CRediT roles would need to be added, and some testing would need to be done to make sure that this operates as expected throughout, as multiple author user groups are not commonly used in OJS and there may be some assumptions built into the code that don't serve CRediT well.

**Comment by ctgraham** [comment]
There is another level of selection of the Role (as opposed to User Group?) within the Submission Metadata, under "Edit Contributor".  The "Contributor Role" here lists any Role with a "Permission Level" as reported in Users & Roles (same as User Group?) of "Author".  This is, I think, where the forum users propose adding a 1:n relationship.

**Comment by willinsky** [comment]
I may not have understood the discussion  above  properly, but I would like to add my support for PKP adopting the CRediT standard. It does mean adding a metadata category "Author Role (CRediT)" selected for each author from the CRediT controlled vocabulary of 14 "roles," which would be in addition existing categories for authors such as "affiliation," but this does not impact OJS user roles or permission levels, and would be article dependent.

**Comment by pmangahis** [comment]
A hosted client has also expressed interest in having this as as a plugin.

**Comment by NateWr** [comment]
I had a discussion with two early career researchers about the CRediT standard and how it should be implemented in our software. The consensus was that it was a) a very good thing and b) authors need an opportunity to declare their own roles as well as dispute other authors' claims.

The main issue is the regular abuse by supervisors and senior academics of authorship attribution, when they claim inappropriate roles on early career researchers' papers. CRediT was seen as a good thing because lots of abuse happens through the ambiguous first author/last author distinctions in disciplines. 

They wanted to see a workflow that countered this abuse:

- The submitting author identifies other contributors but is not asked for their CRediT roles.
- Each contributor is emailed (maybe after passing peer review) and presented with a form where they must tick off the roles they believe they qualify for.
- When all contributors have completed this, the submitting author is asked to confirm or clarify any role attribution before the article is published.
- Editors should the final arbiters of any disputes that arise, and they should be the one's responsible for clarifying what roles should be designated.

**Comment by asmecher** [comment]
@NateWr, just flagging that some of this might be possible to delegate to the ORCID workflow, since it already provides a way to ping secondary contributors so that they can authenticate themselves against that service.

**Comment by nuest** [comment]
I think it would be great to capture author contributions in OJS with CRediT. Just one more related idea (I can try to dig up the original article on that - it's not mine): OJS could show the authors in randomised order or suggest/force alphabetical order on the website, as the real contributions are properly modeled.

**Comment by willinsky** [comment]
@nuest Daniel, we have always to be careful, as platform developers, to support publisher policies and practices, including the promotion of what we see as best practices among those policies and practices. CReDiT has that publisher support at this point, so we can make it an option. Do you see that for randomized or alphabetical author order?

**Comment by fgnievinski** [comment]
may I suggest offering initially simply a free-form field where the lead author can paste the entire contributors' role statement, such as: 

"ABC contributed to conceptualization, data curation, and formal analysis; DEF contributed to funding acquisition and project administration; GHI contributed in supervision and writing."

normally the statement is discussed and composed among team members outside of OJS.

**Comment by fgnievinski** [comment]
also please note that it's becoming more and more common for journals to incorporate the contributors' role statement in the published article, at the bottom of the text, just before References.

**Comment by nuest** [comment]
@willinsky I certainly was not considering the broader perspective of publisher policies in my suggestion. Support for randomisation or other ordering should certainly be optional, and maybe the feature is better suited for a plug-in.

Re. the free-form field, I want to point out that explicit modelling (from the start) is a better basis for future meaningful annotation and publishing of the roles properly in well-defined (or even stanardised) metadata. A free-form field as an alternative to CRediT does have benefits, for example if authors or journals feel overwhelmed or restricted by the formal definitions of CRediT.

**Comment by marcbria** [comment]
The Spanish Foundation for Science and Technology (a national agency) will probably ask all science journals to use CRediT to specify the role of each author.

**Comment by asmecher** [comment]
@marcbria, that's a good incentive to get this launched! But two questions...
- is the distinction between author and translator relevant to their use case? It's the most common distinction editors need (anecdotally) but CRediT doesn't have it.
- what mechanisms will the agency want to use to consume CRediT metadata? (E.g. OAI-PMH, CrossRef, etc)

**Comment by marcbria** [comment]
> is the distinction between author and translator relevant to their use case? It's the most common distinction editors need (anecdotally) but CRediT doesn't have it.

Not yet, but it will in a couple of years.
I understand why CRediT didn't include translators (translation is considered a secondary task in terms of science). 
But in the other hand, according to intellectual property laws a translator is also an author so they need to be included.
What was your plan for this?

> what mechanisms will the agency want to use to consume CRediT metadata? (E.g. OAI-PMH, CrossRef, etc)

The agency won't consume this data yet. It will probably start as a recommendation for journals next year... and till a couple of years from now won't be a mandatory requirement for the journals that like to get the seal of the "ministry of science and technology".

**Comment by asmecher** [comment]
> I understand why CRediT didn't include translators...

It's actually worse than that :)

The "[Writing - Original Draft](https://credit.niso.org/contributor-roles/writing-original-draft/)" CRediT role explicitly *includes* translation work:

> Preparation, creation and/or presentation of the published work, specifically writing the initial draft (including substantive translation).

...which means that if translators are added to the submission as contributors, their contributions will be indistinguishable from authorship.

**Comment by marcbria** [comment]
But in my opinion the key is in the word "substantive". 

We need to think in two kind of translators:

The role of "translator", as it is commonly used in scientific journals, is for a translation that, by definition, is "non-substantive"... that is, that "does not alter the subject matter or essence of the article", because if it did, it would alter the article and would be a bad translation.

This is very different from the role that (usually one of the co-authors) can play when, from the knowledge of the research (and contributing with his/her specialized terminology knowledge or his/her ability to adapt the text to the new regional context, etc.) he/she adds "substance" with his/her translation.

Imagine a group of researchers who write a draft in language other than English: Probably one of them will need to translate that draft into English before been submitted to a journal.

In short, I see no conflict in considering "substantive translation" as authorship, since it is a competence with as much value as writing style.

What I would find strange (in science) is that someone who has made a literal (or non-substantive) translation need to be recognized as an author... although I seem to remember that Spanish intellectual property law (I don't know about other places) does attribute authorship of translation.

I will consult a lawyer specialized in intellectual property and let you know.

**Comment by fgnievinski** [comment]
I've found the following guidance:

"They [translators] should be included in an ‘Acknowledgments’ section with an explanation of their role, *or* they should be included in the author list if appropriate."
https://authorservices.taylorandfrancis.com/editorial-policies/defining-authorship-research-paper/

The key, then, is appropriateness.  Different organizations offer various criteria for authorship, e.g.:
https://publicationethics.org/files/COPE_DD_A4_Authorship_SEPT19_SCREEN_AW.pdf]
https://www.ease.org.uk/wp-content/uploads/2018/11/doi.10.20316.ESE_.2018.44.e1.pdf
https://www.ismpp.org/assets/docs/Inititives/amwa-emwa-ismpp%20joint%20position%20statement%20on%20the%20role%20of%20professional%20medical%20writers_january%202017.pdf

So it seems we can't escape supporting at least two cases: translators only acknowledged and translators as coauthors.

A translator hired by the original authors of a complete but unpublished work would more likekly be only acknowledged by the authors, with no need for CRediT attribution.

Now, a person fluent in the two languages who was deemed by the original authors of an unpublished work to meet the criteria for coauthorship in a future derived work would get CRediT attribution in the contributor role of "Writing - Original Draft" and/or "Writing – review & editing".

**Comment by fgnievinski** [comment]
The two cases above involved translations of unpublished works. There's actually a third case: a translator hired or volunteered without collaboration of the original authors of a published work. In that case, translator could be mentioned in a footnote, but there is no consent from the original authors for the translator to be listed as a coauthor or to be mentioned in an acknowledgement. So, again, no need for CRediT attribution.

Of course, citation styles do allow for crediting the translator, though clearly separated from authors, not unlike how volume editors can be credited in the citation:
https://apastyle.apa.org/blog/citing-translated-works
https://docs.citationstyles.org/en/stable/specification.html#name-variables

So, I'd suggest splitting in a separate issue the formal crediting of translators who do not meet the criteria for coauthorship. 

Maybe OJS could even borrow from OMP in the way non-author contributor roles are handled:
https://docs.pkp.sfu.ca/learning-omp/en/catalog-management#highlight-author-who-is-an-editor-of-the-volume

**Comment by ajnyga** [comment]
Just adding a link to my comment in another issue because it is also related to this.

I think a lot of our problems derive from the fact that we try to use the same taxonomy of user roles both for:
1. Handling user permissions in the system, ie. what can a user do with a specific role
2. In article metadata desrcive the contributors role

**I think we should just separate these two very different things completely.**

https://github.com/pkp/pkp-lib/issues/7223#issuecomment-1490106394

**Comment by nils-stefan-weiher** [comment]
Hi  @ajnyga ,

>   Separate these two very different ways of using our current role taxonomy: Continue to have the current approach for user permissions BUT create a separate Contributor Role taxonomy just for metadata purposes, something we can map to different metadata schemas (ORCID, Crossref etc.). Basically determine a set of roles that work in most cases, like Author, Chapter Author, Editor, Translator etc.

This is also relevant for the citationStyleLanguage Plugin, because at the moment you have to set up the Contributor roles specifically to distintiguish translator, chapter authors, etc.

**Comment by ajnyga** [comment]
Thanks, yes we have recognized this problem

**Comment by asmecher** [comment]
@ctgraham [raised the issue](https://github.com/pkp/pkp-lib/issues/857#issuecomment-153367307) of degree of contribution. This was also raised in https://github.com/pkp/pkp-lib/issues/6395.

CRediT does specify a vocabulary for this, and I suggest we implement it when integrating CRediT into the application.

From [CRediT documentation](https://credit.niso.org/implementing-credit/):
> Degree of Contribution Optional – Where multiple individuals serve in the same role, the degree of contribution can optionally be specified as ‘lead’, ‘equal’, or ‘supporting’;

**Comment by altos-group** [comment]
### Subject: Proposal for Integrating CRediT Standard with PKP Library via CredIT Forge

Dear fellow researchers,

We are the Casrai CredIT R&D team writing to propose an innovative solution for the integration of the CRediT (Contributor Roles Taxonomy) standard within the PKP library, specifically addressing the issues above. As you're aware, accurately recognizing individual contributions is increasingly important in scholarly communications. Our tool, CredIT Forge, leverages the powerful Llama 2 AI model from Meta, trained specifically on contributor taxonomy, to automate and streamline the process of contributor attribution.

### Key Features of CredIT Forge:

Automated Taxonomy Contribution: Utilizes Llama 2 AI to intelligently classify and attribute various scholarly contributions based on the CRediT taxonomy.

Flexible Input & Output Formats: Accepts research summaries in natural language, LaTeX, TikZ, and other metadata, and outputs a comprehensive PDF of the contribution graph and other formats as needed.

Easy Integration: Simple endpoint integration using PHP as illustrated, allowing for quick adoption and scalability within existing PKP infrastructure.

### Advantages to the PKP Community:

Standardized Contributions: Ensures a consistent, industry-wide approach to contributor attribution, aligning with the CRediT initiative by CASRAI.

Enhanced Recognition: Provides a detailed, accurate account of each contributor's role, promoting fair and transparent acknowledgment.

Scalability & Support: Designed with future expansion in mind, ensuring the tool remains adaptable and supportive of evolving scholarly communication needs.

While we recognize that ongoing maintenance and updates are not without cost, the unique capabilities and the potential for significant positive impact on the academic community make CredIT Forge a worthwhile investment. As members of the CASRAI community, we are committed to the original taxonomy and its goals of improving research outputs' clarity and usability.

We would be honored to further discuss how CredIT Forge can specifically benefit the PKP community and facilitate a smoother, more integrated approach to adopting the CRediT standard. Our dedicated team is ready to support the integration process and ensure a seamless transition for all stakeholders involved.

For a deeper understanding of our tool and its potential, please visit our GitHub repository: https://github.com/CASRAI-CRedIT/CRedIT

Thank you for considering this proposal. We are excited about the possibility of contributing to the PKP library and enhancing the recognition of scholarly contributions through the adoption of CRediT. For any necessary use cases needed before adoption, please reach out to the team, we are working diligently to solve contributor credit taxonomy challenges. We need your help and feedback.

**Best Regards,

Gabriel Nieves, 
CTO
research@casrai.org
Casrai CredIT Research & Development Team**

**Comment by CASRAI-CRedIT** [comment]
### Subject: Proposal for Integrating CRediT Standard with PKP Library via CredIT Forge

Dear fellow researchers,

We are the Casrai CredIT R&D team writing to propose an innovative solution for the integration of the CRediT (Contributor Roles Taxonomy) standard within the PKP library, specifically addressing the issues above. As you're aware, accurately recognizing individual contributions is increasingly important in scholarly communications. Our tool, CredIT Forge, leverages the powerful Llama 2 AI model from Meta, trained specifically on contributor taxonomy, to automate and streamline the process of contributor attribution.

### Key Features of CredIT Forge:

Automated Taxonomy Contribution: Utilizes Llama 2 AI to intelligently classify and attribute various scholarly contributions based on the CRediT taxonomy.

Flexible Input & Output Formats: Accepts research summaries in natural language, LaTeX, TikZ, and other metadata, and outputs a comprehensive PDF of the contribution graph and other formats as needed.

Easy Integration: Simple endpoint integration using PHP as illustrated, allowing for quick adoption and scalability within existing PKP infrastructure.

### Advantages to the PKP Community:

Standardized Contributions: Ensures a consistent, industry-wide approach to contributor attribution, aligning with the CRediT initiative by CASRAI.

Enhanced Recognition: Provides a detailed, accurate account of each contributor's role, promoting fair and transparent acknowledgment.

Scalability & Support: Designed with future expansion in mind, ensuring the tool remains adaptable and supportive of evolving scholarly communication needs.

While we recognize that ongoing maintenance and updates are not without cost, the unique capabilities and the potential for significant positive impact on the academic community make CredIT Forge a worthwhile investment. As members of the CASRAI community, we are committed to the original taxonomy and its goals of improving research outputs' clarity and usability.

We would be honored to further discuss how CredIT Forge can specifically benefit the PKP community and facilitate a smoother, more integrated approach to adopting the CRediT standard. Our dedicated team is ready to support the integration process and ensure a seamless transition for all stakeholders involved.

For a deeper understanding of our tool and its potential, please visit our GitHub repository: https://github.com/CASRAI-CRedIT/CRedIT

Thank you for considering this proposal. We are excited about the possibility of contributing to the PKP library and enhancing the recognition of scholarly contributions through the adoption of CRediT. For any necessary use cases needed before adoption, please reach out to the team, we are working diligently to solve contributor credit taxonomy challenges. We need your help and feedback.

**Best Regards,

Gabriel Nieves, CTO
research@casrai.org
Casrai CredIT Research & Development Team**

**Comment by TAC-NISO** [comment]
To avoid any confusion, the new incarnation of CASRAI is not involved with, nor are they maintainers of the CRediT standard, which was transferred to NISO in 2019. The CRediT standard is published as ANSI/NISO Z39.104-2022, CRediT, Contributor Roles Taxonomy - for a link to the current version: https://www.niso.org/publications/z39104-2022-credit

**Comment by CASRAI-CRedIT** [comment]
> To avoid any confusion, the new incarnation of CASRAI is not involved with, nor are they maintainers of the CRediT standard, which was transferred to NISO in 2019. The CRediT standard is published as ANSI/NISO Z39.104-2022, CRediT, Contributor Roles Taxonomy - for a link to the current version: https://www.niso.org/publications/z39104-2022-credit

Hi,

Thank you for your attention to the developments in the research information management industry. I want to clarify a few points to avoid any confusion and to highlight our ongoing commitment to supporting researchers and institutions.

Firstly, it's important to understand that our organization, distinct from the previous incarnation of CASRAI, is dedicated to providing practical, usable tools for the research community. We acknowledge that the CRediT standard is now under the stewardship of NISO, as ANSI/NISO Z39.104-2022, and we respect their role in this space. Our focus, however, is on creating solutions that researchers and institutions can directly benefit from.

In our efforts to support the Open Journal Systems (OJS) community, we have developed a service that we believe addresses many of the current needs in research information management. Our interactions with developers, including @asmecher, are aimed at fostering collaboration and integration, though we understand the challenges that come with such endeavors.

We are aware that the incorporation of the CASRAI dictionary of elements into existing systems is complex and might not align with the current development priorities of some platforms. However, our service has been designed to be adaptable and user-friendly, providing a valuable resource for those who choose to utilize it.

As a non-profit organization, we are driven by a mission to serve the research community, not by profit motives. The service we provide is free of charge, though it does require significant resources to maintain. We believe in the power of collaboration and shared knowledge, and we invite you and others in the community to explore how our tools can be of benefit.

We understand that discussions about management and standardization are important, but our primary focus is on delivering practical solutions. We encourage you to consider our service as an option in your toolkit. It is designed with the user in mind, aiming to streamline processes and enhance the research information management landscape.

Your consideration of our efforts is greatly appreciated, and we remain open to feedback and collaboration. We are all part of the same community, striving towards the common goal of advancing research and its management.

We have tried to reach out to NISO ad nauseam to no avail. We welcome open communication between our organizations. @asmecher does not want to work with us at this point, we will continue to develop our metadata OJS plugin and provide it free of use on various platforms. For updates on this visit us at casrai.org/credit .

Best Regards,
Gabriel Nieves
Consortia Advancing Standards in Research Administration Information

**Comment by jalperin** [comment]
It would be great to add a way for users to be able to access the description of the roles from within the UI.

**Comment by ajnyga** [comment]
You are mostly thinking of the readers and the article landing page, right?

**Comment by jalperin** [comment]
No, authors entering the information

**Comment by ajnyga** [comment]
Ok, I can see that it would be beneficial in both use cases.

**Comment by asmecher** [comment]
I would much rather link out to a 3rd-party resource than duplicate the descriptions inside OJS -- however the https://credit.niso.org/ website is not-great. (And actually kind of broken at the moment -- links contain IP addresses that then redirect to the domain name!) There is no provision for multilingualism e.g. in the URLs. 

I'm already not satisfied with maintaining our own files for the translation of terms, and local copies of the descriptions would compound this.

I do see that there's a [relatively new repo](https://github.com/contributorshipcollaboration/credit-translation/tree/main?tab=readme-ov-file) out there to resolve this outside the plugin and I think that's the right approach (since the NISO CRediT group isn't endorsing an approach to translation). It appears to have both names of roles and descriptions translated -- though there are some key translations missing (e.g. Spanish).

**Comment by jalperin** [comment]
The first contributors to that repo are our contacts who are coordinating the translation of the plugin. Is the right approach, then, for us to pull in terms and descriptions from that repo?

**Comment by asmecher** [comment]
We currently get our English list of roles (and main list of URIs) [from the `jats-schematrons` repo](https://github.com/JATS4R/jats-schematrons/blob/master/schematrons/1.0/credit-roles.xml), which is included into the CRediT plugin as a submodule. We could also add [credit-translation](https://github.com/contributorshipcollaboration/credit-translation) as a submodule.

In addition to a bit of coding, we'd need to move the [Spanish translation](https://github.com/asmecher/credit/tree/main/translations) into that repo (in the required form); ours contains just the CRediT roles, not the descriptions. I don't know if they've got an expression of interest from someone working in Spanish.

**Comment by jalperin** [comment]
Not sure if they have someone for Spanish. @alexholcombe should be able to answer that. 

Would you need to move the Spanish XML list? Or remove it and use the English one in combination with the translation of terms? I note that the URIs in the Spanish one still point to the English-language URIs, and that there are no `normalized-term` attributes.

**Comment by alexholcombe** [comment]
About Spanish, we have people working on Spanish, which will be added [here](https://github.com/contributorshipcollaboration/credit-translation/tree/main/translations), but it will be at least a month before the translation is finished (we have a quality control process involving back-translation and two people contributing).

About the CRediT description text , e.g. "Ideas; formulation or evolution of overarching research goals and aims." which is the explanation of what "Conceptualization" means, NISO seems to have no machine-readable way of pulling the description text. About the jats-schematrons repo, we are trying to find out whether there could be recommended approach for getting the text through that, and also implementing the translations.

@CASRAI-CRedIT, thanks for explaining your recent efforts in this space. But when I go to the https://casrai.org/credit link you provided, I'm getting a "404 Not found".

**Comment by asmecher** [comment]
@jalperin / @alexholcombe, about Juan's earlier suggestion:

> It would be great to add a way for users to be able to access the description of the roles from within the UI.

While the [credit-translation repo](https://github.com/contributorshipcollaboration/credit-translation) does include space for translations of both roles and descriptions, the [jats-schematrons](https://github.com/JATS4R/jats-schematrons/blob/master/schematrons/1.0/credit-roles.xml) (where we're getting English text and the overall list) only contains English names, not descriptions. And I worry that the translation repo could be quite sensitive to tiny discrepancies between the two.

@alexholcombe, what do you think about changing your repo from using an English-centric structure like (from the German translation)...
```json
{
   "Conceptualization": {
      "description": "Ideas; formulation or evolution of overarching research goals and aims.",
      "id": "8b73531f-db56-4914-9502-4cc4d4d8ed73",
         "translation": {
         "name": "Konzeptualisierung",
         "description": "Ideen; Formulierung oder Entwicklung der übergeordneten Forschungsziele."
      }
   },
   ...
}
```
...to instead using a URI-centric structure like this?
```json
{
    "http://credit.niso.org/contributor-roles/conceptualization/": {
        "name": "Konzeptualisierung",
        "description": "Ideen; Formulierung oder Entwicklung der übergeordneten Forschungsziele."
    },
    ...
}
```
Then the English-language form could also be added as a translation... For example, `en.json`:
```json
{
    "http://credit.niso.org/contributor-roles/conceptualization/": {
        "name": "Conceptualization",
        "description": "Ideas; formulation or evolution of overarching research goals and aims."
    },
    ...
}
```
...and there could be a single descriptor to list them all (including with IDs, I suppose, though I'm not sure where they're used):
```json
{
    "http://credit.niso.org/contributor-roles/conceptualization/": {
        "id": "8b73531f-db56-4914-9502-4cc4d4d8ed73",
        ...
    },
    ...
}
```

**Comment by marcbria** [comment]
> I'm already not satisfied with maintaining our own files for the translation of terms, and local copies of the descriptions would compound this.

I was passing by... this thread, just to comment that, I strongly agree in this context.

I mean, that +100 that we do NOT maintain our own translations because this is a standard and there is no provision for our needs to differ from the general ones.

**Comment by asmecher** [comment]
I've proposed a conversion of the [credit-translation](https://github.com/contributorshipcollaboration/credit-translation) repo as described above here: https://github.com/contributorshipcollaboration/credit-translation/pull/22

**Comment by kmccurley** [comment]
In case you are not aware, crossref has initiated a [request for feedback](https://community.crossref.org/t/request-for-feedback/6610) about a proposal for extending CRediT roles. Of course this will remind us that ["The nice thing about standards is that there are so many to choose from"](https://xkcd.com/927/).

**Comment by alexholcombe** [comment]
Overall, it seems the CRediT plugin is now working but can anyone confirm they have actually deployed at and are using it at their journal? 

About core (non-plugin) CRediT support, elsewhere I saw an update from Mark Huskisson that this will be happening thanks to development by the teams at TSV (Federation of Finnish Learned Societies) and the CRAFT OA project. 

About the CRediT full text descriptions and translations into other languages, thanks @asmecher, as you know we changed our CRediT translation schema into the format you suggested! So the JSON files for the CRediT translations are [here](https://github.com/contributorshipcollaboration/credit-translation/tree/main/translations) (and the human-readable versions are [here](https://contributorshipcollaboration.github.io/projects/translation/completed/)).

**Comment by alexholcombe** [comment]
> Sorry Alex but why Catalan is missing in this list? Translation work was done some months ago.
> 
> If you need anything, just ask for it.

Thanks for pointing this out @marcbria! The last step that is yet to be done for Catalan is to copy/paste all your translated text into the appropriate parts of a JSON file, which temporarily slipped off our task list but I have now put it back on the agenda for the next hackathon 22 Jan (or if you have time, [here](https://github.com/contributorshipcollaboration/credit-translation/tree/main/translations) you can see the JSON format).

**Comment by bozana** [comment]
Hi @Tribunal33, everything is merged here. Would you like to take a look at this new feature, and then close the issue?
Thanks a lot!

**Comment by Tribunal33** [comment]
So far this is looking great.  I see a few issues I'm not sure about what is expected.  @bozana I was seeing an issue with Quick Submit plugin where the contributor's wouldn't load.  I thought this might be related but looks like it has been fixed.  I'm now not seeing the new CrediT inside this 'add contributor' form.  Is this expected?

~1.  Quick Submit plugin 'Add Contributor' form.  Should we have this working for that plugin as well?~ won't fix
2. Currently when you select two of the same roles it appears in the list but when saved there is only the last one that get's saved.
3. Inconsistencies with similar table (invite to a role)
a. `Remove Role` button style is different
b. Ability to add the same role doesn't have same grey out logic for previous selected roles
 

<img width="2309" height="1231" alt="Image" src="https://github.com/user-attachments/assets/bfe9d240-2f13-4b55-9043-9ed0c8449368" />

**Comment by bozana** [comment]
Hi @Tribunal33,
I think we should not invest more time into QuickSubmit plugin -- the quick submission should be implemented in another way as soon as possible -- I am not sure but I believe for 3.6 🤔 

It is correct that one role can be saved only once, but I could maybe try to consider this in the UI as well, similar to the invitation form. Let me try... :-)

Regarding the button "Remove" I need to ask @jardakotesovec 😅

**Comment by kmccurley** [comment]
Before you deprecate the QuickSubmit plugin you might want to make sure the REST API is a viable alternative. Several of us tried to create submissions that way and ran into a brick wall. See [this post](https://forum.pkp.sfu.ca/t/is-there-any-documentation-on-how-to-fully-create-a-submission-via-rest-api/74730/16). In my case the last step was that I couldn't figure out what the userGroupId was supposed to be when creating a contributor for a submission. The documentation also says the encoding for creating a submission should be `application/json`, but in fact the UI sends data encoded as `x-www-form-urlencoded` and `application/json` did not work. I was working on `stable-3_4_0` at the point when I gave up. It appears that the documentation for the REST API is incomplete and the UI doesn't exactly use it.

Our journals do not use OJS for the submission, review, copy editing, or production steps - only the DOI registration and hosting. We were trying to create publications once they had passed all those phases. This usage may violate some underlying assumptions in the codebase, but is [suggested here](https://docs.pkp.sfu.ca/journal-policies-workflows/en/alternative-uses#using-ojs-for-publishing-only). Is there some documentation on the plan for the QuickSubmit plugin? I would honestly prefer to use the API because we typically insert 20-30 articles for an issue. Someone should work though that entire flow and maybe provide an example.

**Comment by asmecher** [comment]
@kmccurley , the QuickSubmit plugin will be rewritten -- hopefully but not definitely in the 3.6 development cycle. We are avoiding changes to it until then because it'll just build further tech debt, but the plugin does have a future.

**Comment by Tribunal33** [comment]
@bozana thank you for clarifying the QuickSubmit issue.  I marked it as `won't fix` since it is out of scope for now.  Let me know once the other issues have been considered and I will retest again.

**Comment by Tribunal33** [comment]
@bozana I was just checking again with the combined effort of the new submission file upload moving to step 1.  Noticed that when  a contributor is automatically added their `Affiliation` is added if it exists but the Credit role will not be.  Wondering if there is a way to set the Credit Role to the profile similarily to Affiliation?  The following items are currently under your consideration:
- [x] Grey out of multiple Credit Roles (similar to Invite to a role)
- [x] Remove Role button not matching invite to a role table button
- [ ] ~~Auto-adding contributor to new submission will bypass the CrediT functionality.~~

**Comment by bozana** [comment]
Hi @Tribunal33, I co not think CRediT can be applied to the user profile -- a user can have different roles in different articles -- they describe the role and level of activity of a contributor in a publication/article.

**Comment by bozana** [comment]
hi @Tribunal33, the two points regarding UI are merged, so your turn again :-)
Thanks a lot!

**Comment by Tribunal33** [comment]
Thanks @bozana and @jardakotesovec.  This passes QA on main.  I have checked the 2 other UI issues and they have been resolved and in general the contributor CrediT roles feature is working.  What I could see is that New submission Contributors and Publication > Contributors sections had this new field table.  Out of scope are quickSubmit plugin.

Here are the tests I added to the regression doc with more information on expected outcomes in the document:
1. Test if new submission Add Contributor has the new Credit Support field
2. Test that Add new Role button adds default role
3. Test adding a new role happy path
4. Test adding a new role without a degree
5. Test adding multiple roles
6. Test removing role button
7. Test removing role button for all roles
8. Test Editing a contributor with CrediT roles added
9. Test if CrediT Roles added during submission process are saved in submission publication data
10. Test if Publication > Contributors Add new contributor has Credit Support field
11. Test CrediT roles in publication > Contributors

**Comment by alexholcombe** [comment]
A student who worked with us on translating CRediT, Dunja Micunovic, volunteered  to log in and try out the sandbox provided by @ajnyga . Good news, she found it easy to use:

> As far as I’m concerned (and considering that I’d never used OSJ system before), it looks good and easy to operate. It reminded me a bit of ORCID, so I figured it out. Is this supposed to be like an easy site that researchers can use? For example, are these Active submissions the ones submited for different journals, and then they are all listed here for easier organizing? I would love to hear more about all of it.
> 
> However, from what I’ve seen, it seems good!

**Comment by ajnyga** [comment]
Great to hear thanks for the feedback.

She was probably looking at the editor's view to the system where the editor can see all the existing submissions and also, if needed, edit the metadata.

The same form for adding contributors is available also for authors when they send their own article manuscript to the system. In the submission form they provide the article metadata including the contributor metadata and CRediT roles. Of course this data can then be checked and further edited by the editors during the publication process.

**Comment by jonasraoni** [comment]
I've just skimmed the PR, so please confirm to me:
1. I'd say the `plugins/generic/credit` submodule isn't needed anymore and should be removed and archived.
2. I haven't seen a migration for the data which was previously managed by the plugin.

**Comment by alexholcombe** [comment]
It makes sense to me that the plugin will no longer be needed once this version is released. 
About the data managed by the plugin, I suspect very few journals ended up using the plugin, in part because it hasn't been available for very long and IIRC, the plugin gallery wasn't very salient and was being re-developed. 
Regarding migrating the data, to make sure I understand correctly, I suppose that would include data from all articles, both published, and still in process, for which someone used the CRediT plugin, correct? I remember that at one point I discussed with Alec adding a checkbox to the plugin for reporting to OJS the plugin usage (which I don't think was implemented), so I suppose there is no way to know whether someone was actually using it, which is unfortunate. One reason I ask is that we received a grant from the Experiment foundation in part to investigate whether adoption of CRediT is associated with incorporating a broader range of co-authors, e.g. geographically or just more co-authors. When this version is released with it part of the core, I suppose there still won't be any 'pings' going to PKP, but of course we can scrape OJS journals to see how which of them have CRediT information.

Just so everyone can see what's been done, I'm attaching a screenshot of what the CRediT feature looks like on the submission side. It looks quite nice!
<img width="1179" height="519" alt="Image" src="https://github.com/user-attachments/assets/b90a9efd-5f90-4a62-8ab4-4304f0dab917" />

**Comment by alexholcombe** [comment]
About the degree of contribution feature, as you know it is optional, and many submitting authors will not want to use it. I think it should be made more clear in the user interface that it is optional, as currently it sounds mandatory ("Select the CRediT roles of the contributor and the degree of contribution"). I know that the absence of the red asterisk should convey that it is optional, but I still think that optional should be added somewhere, perhaps to make the sentence as follows: "Select the contributor's CRediT roles and, optionally, their degree of contribution."  One reason I am so attuned to this is because this degree thing can be very confusing for some, partly because almost no one has any experience with it, and partly because the options are not real coherent (e.g., when shoul done use 'equal' and when should one use 'supporting'?).

**Comment by ajnyga** [comment]
> I've just skimmed the PR, so please confirm to me:
> 
> 1. I'd say the `plugins/generic/credit` submodule isn't needed anymore and should be removed and archived.
> 2. I haven't seen a migration for the data which was previously managed by the plugin.

Thanks for noticing this. We could maybe add a migration although the amount of the data might be small. The migration should be quite straigthforward.

**Comment by bozana** [comment]
Hi all, I opened a new issue that should consider data migration: https://github.com/pkp/pkp-lib/issues/11914
I have fully forgotten it. Thanks @jonasraoni for catching this!
@ajnyga, may I assign you and Jyrki to the new issue, to work on it?

**Comment by ajnyga** [comment]
Go ahead and assign us
